project/src/main/java/com/google/sps/data/UserFeedback.java

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.Optional;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+
+/**
+ * Represents the feedback that users enter about the recommendations they were given.
+ */
+@AutoValue
+public abstract class UserFeedback {
+
+  /** @return a list of IDs of the places that were recommended to the user. */
+  public abstract ImmutableList<String> placesRecommendedToUser();
+
+  /** @return the ID of the place the user chose, if any. */
+  public abstract Optional<String> placeUserChose();
+
+  /**
+  * @return true if the user requested to try again and get more recommendations (implying they
+  * weren't satisfied with the recommendations they recieved).
+  */
+  public abstract boolean userTriedAgain();
```

Comment:
Can a user try again just once? What if the user clicked it multiple times (IMO that should be taken as a satisfactory factor looking forward), but even for now you can keep a "retriesCount" instead. WDYT?

Reply:
I don't think this is information we would want to use for our current purposes, bc the number of times a user tried again says more about the user than about his thoughts about the specific current place. maybe we'll want to add this in the future but I don't think adding this now will help with the current purposes.